### PR TITLE
Always move in the right direction for PageUp/PageDown

### LINKED ISSUE #4213: PageUp/PageDown don't work in single line CodeMirror
1. Go to https://codemirror.net/demo/resize.html
2. Set the content of the editor to be `123`
3. Put your cursor at the end of the editor and press page up.
    * The cursor unexpectedly stays at the end of the editor
4. Set the content of the editor to be two lines:
```
123
456
```
5. Put your cursor at the end of the first line and press page up
    * The cursor moves to the start of the editor as expected.

This appears to be dependent on the CSS applied to the editor. It doesn't happen in the main editor on the CodeMirror.net home page.

Tested in Chrome and Firefox.
Thanks!

### PR DESCRIPTION
**Pull request by JoelEinbinder**
Closes #4213

### CODE REVIEW

File: lib/codemirror.js [hunk 1]
```diff
@@ -4907,7 +4907,9 @@
     var doc = cm.doc, x = pos.left, y;
     if (unit == "page") {
       var pageSize = Math.min(cm.display.wrapper.clientHeight, window.innerHeight || document.documentElement.clientHeight);
-      y = pos.top + dir * (pageSize - (dir < 0 ? 1.5 : .5) * textHeight(cm.display));
+      var moveAmount = pageSize - (dir < 0 ? 1.5 : .5) * textHeight(cm.display);
+      moveAmount = Math.max(moveAmount, 1);
```
Comment (marijnh) [inline lib/codemirror.js]: Why do you clip to 1, and not 0?

**Comment by JoelEinbinder** [comment]
If it clipped to 0, a page up would have no effect. I was clipping to 1, but I noticed you used 3 below so I switched to that. I added a test.

**Comment by JoelEinbinder** [comment]
Let me know if there's anything else you'd like me to tweak.

**Comment by marijnh** [comment]
Looks great. Merged as e59d75a

**Pull request closed without merging**